src/test/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBamIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+package org.broadinstitute.hellbender.tools.spark;
+
+import htsjdk.samtools.BAMIndex;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.seqdoop.hadoop_bam.SplittingBAMIndex;
+import org.seqdoop.hadoop_bam.SplittingBAMIndexer;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+
+public class CreateIndexForHadoopBamIntegrationTest extends CommandLineProgramTest{
+
+    private final File UNSORTED_BAM = getTestFile("count_reads.bam");
+    private final File SORTED_BAM = getTestFile("count_reads_sorted.bam");
+
+    @DataProvider(name="indexable")
+    public Object[][] getIndexableFiles(){
+        return new Object[][]{
+                {UNSORTED_BAM},
+                {SORTED_BAM}
+        };
+    }
+
+    @Test(dataProvider = "indexable")
+    public void testCreateSplittingIndex(File bam) throws IOException {
+        File splittingIndex = getTempIndexFile();
+        splittingIndex.delete();
+        Assert.assertFalse(splittingIndex.exists());
+        ArgumentsBuilder args = getInputAndOutputArgs(bam, splittingIndex)
+           .add("--"+ CreateIndexForHadoopBam.SPLITTING_INDEX_GRANULARITY_LONG_NAME).add("1");
+        this.runCommandLine(args);
+        assertIndexIsNotEmpty(splittingIndex);
+        //checked in index created with
+        // java -cp target/hadoop-bam-7.4.1-SNAPSHOT-jar-with-dependencies.jar org.seqdoop.hadoop_bam.SplittingBAMIndexer  1 <filename>
+        IOUtil.assertFilesEqual(splittingIndex, new File(bam.toPath() + SplittingBAMIndexer.OUTPUT_FILE_EXTENSION));
```

Comment:
extract the the name of the checked in file as a constant (or at least a variable with a name that makes it clear that this is a pre-created file)